docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,224 @@
+---
+title: Upgrade to Prefect 3
+description: Learn how to upgrade from Prefect 2 to Prefect 3.
+---
+
+Some patterns supported by Prefect 2 are no longer supported with Prefect 3, or are enabled differently. 
+This guide covers changes that you may need to make to ensure your flows keep flowing.
+
+## Module locations and names
+
+Some modules have been renamed, reorganized, or removed for clarity. Objects imported from changed modules will continue to work until their deprecation period ends, but will raise deprecation warnings. 
+You must update these modules before the six month deprecation period ends:
+
+| Prefect 2 | Prefect 3 | Notes                                 |
+| --------- | --------- | ------------------------------------- |
+|           |           |                                       |
+
+## Asynchronous task execution behavior
+
+Prefect 2 abstracts away certain task execution concerns, which is convenient in some cases, but comes at the expense of unexpected behavior in other cases. 
+Prefect 3 defaults to running tasks as they would run as normal Python code, and requires you to be more explicit when you would like to modify task execution behavior. 
+This results in clearer behavior with fewer surprises.
+
+In Prefect 3, all workflow code, including task functions, runs on the main thread. This means:
+
+- Python code that is not explicitly defined as asynchronous will execute synchronously, as normal Python code does.
+- Non-thread safe objects, such as database connections, can be shared globally within a workflow.
+
+To execute tasks asynchronously in separate threads, you must either write asynchronous Python functions or use the `Task.submit` method. 
+Prefect 3 encourages conventional asynchronous Python programing principles. 
+Updating your workflows to bring them in line with these principles requires code changes. 
+Here are some examples:
+
+**Synchronous tasks and flows cannot be called from asynchronous tasks or flows as they could in Prefect 2.**
+
+<CardGroup cols={2}>
+  <Card title="Prefect 2">
+    ```python
+    @task
+    async def my_task():
+            ...
```

Comment:
this indention looks too deep to me

Reply:
lol agreed I shouldn't have accepted that change. I'll revise it.